fastplotlib/graphics/features/_mesh.py

```diff
@@ -116,3 +116,91 @@ def set_value(
 
         event = GraphicFeatureEvent(type=self._property_name, info={"value": value})
         self._call_event_handlers(event)
+
+
+def surface_data_to_mesh(data: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
```

Comment:
I made this method by simplifying your [x, y, z] -> positions, indices: https://github.com/fastplotlib/fastplotlib/blob/c93b13f0df37015558be93794ebd45611a615154/fastplotlib/layouts/_graphic_methods_mixin.py#L534-L625